Lib/codeop.py

```diff
@@ -69,9 +69,7 @@ def _maybe_compile(compiler, source, filename, symbol):
                 # XXX: RustPython; support multiline definitions in REPL
                 # See also: https://github.com/RustPython/RustPython/pull/5743
                 strerr = str(e)
-                if source.endswith(":") and "expected an indented block" in strerr:
-                    return None
-                elif "incomplete input" in str(e):
+                if isinstance(e, _IncompleteInputError):
```

Comment:
Does this work? Then partially reverting #5743 to the original code will be better, including the upper comment.
```suggestion
                if "incomplete input" in str(e):
```

Reply:
That doesn't work, since `_IncompleteInputError` doesn't necessarily have that string in the error. I've updated the comment and changed this block to just match python 3.15's implementation at least.